### PR TITLE
[aptos-cli] Voting improvements

### PR DESCRIPTION
**Pull request by gregnazario**
### Test Plan
```
aptos governance --help
aptos-governance 0.2.4
Tool for on-chain governance

This tool allows voters that have stake to vote the ability to propose changes to the chain, as well
as vote and execute these proposals.

USAGE:
    aptos governance <SUBCOMMAND>

OPTIONS:
    -h, --help
            Print help information

    -V, --version
            Print version information

SUBCOMMANDS:
    execute-proposal
            Execute a proposal that has passed voting requirements
    help
            Print this message or the help of the given subcommand(s)
    propose
            Submit proposal to other validators to be proposed on
    vote
            Submit a vote on a current proposal
```

New proposal:
```
 aptos governance propose --script-path <SCRIPT_PATH> --framework-git-rev <FRAMEWORK_GIT_REV> --metadata-url <METADATA_URL> --pool-address <POOL_ADDRESS>
```

Voting
```
aptos governance vote --proposal-id <PROPOSAL_ID> --pool-address <POOL_ADDRESS> "--yes or --no"
```

Executing:
New proposal:
```
 aptos governance execute-proposal --script-path <SCRIPT_PATH> --framework-git-rev <FRAMEWORK_GIT_REV> --args ... --type-args ...
```

Propose:
```
aptos governance propose --pool-address $POOL_ADDRESS --metadata-url "https://mocki.io/v1/5714369f-d2d6-4987-83a2-2f3f4c2b5d38" --profile greg --framework-git-rev $FRAMEWORK_GIT_REV --script-path $SCRIPT_NAME
Proposal:
	Title:Hello Title -- Lorem ipsum dolor sit amet
	Description:Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
	Source code URL:https://github.com/some-repo/my-move-update
	Discussion URL:https://forum.aptoslabs.com/proposals/upgrade-potato-module-v3
	Metadata Hash: 706b43a2
	Script Hash: 523e25d6
Do you want to submit this proposal? [yes/no] >
yes
{
  "Result": {
    "proposal_id": 12,
    "transaction_hash": "05645f9b4b57686f5c38d0e1669036e37fa957e43b65e5a496ad97325f2de372",
    "transaction_version": 18449189,
    "gas_used": 20,
    "gas_price_per_unit": 1,
    "sequence_number": 4,
    "vm_status": "Executed successfully"
  }
}
```

Vote:
```
aptos governance vote --proposal-id 12 --pool-address $POOL_ADDRESS --no --profile greg
Are you sure you want to vote No [yes/no] >
yes
{
  "Result": {
    "type": "user_transaction",
    "version": "18497985",
    "hash": "0xc475e176dcdae8feb7cbdfda13f3828af9daaabe1a59c1260535d2
```

Execute:
```
aptos governance execute-proposal --script-path $SCRIPT_NAME --framework-git-rev $FRAMEWORK_GIT_REV --profile greg -
-args u64:10
{
  "Result": {
    "changes": [
      {
        "address": "0000000000000000000000000000000000000000000000000000000000000001",
        "data": {
          "allow_validator_set_change": true,
          "maximum_stake": "100000000000000000",
          "minimum_stake": "100000000000000",
          "recurring_lockup_duration_secs": "864000",
          "rewards_rate": "1902",
          "rewards_rate_denominator": "1000000000",
          "voting_power_increase_limit": "50"
        },
        "event": "write_resource",
        "resource": "0x1::staking_config::StakingConfig"
      },
      {
```

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3075)

### CODE REVIEW

File: crates/aptos/src/governance/mod.rs [hunk 1]
```diff
@@ -160,9 +160,13 @@ pub struct SubmitVote {
...
 
-    /// Vote choice. True for yes. False for no.
-    #[clap(long)]
-    pub(crate) should_pass: bool,
+    /// Vote yes on the proposal
+    #[clap(long, group = "vote")]
+    pub(crate) yes: bool,
+
+    /// Vote no on the proposal
+    #[clap(long, group = "vote")]
```
Comment (banool) [inline crates/aptos/src/governance/mod.rs:168]: Let's add `conflicts_with`

**Review by banool — CHANGES_REQUESTED** [review]

File: crates/aptos/src/governance/mod.rs (hunk 1 above)
Comment (banool) [reply crates/aptos/src/governance/mod.rs:168]: Or is this what `group` is for?

**Review by banool — APPROVED** [review]
Seems like `group` indeed does that, accepting.

File: crates/aptos/src/governance/mod.rs (hunk 1 above)
Comment (gregnazario) [reply crates/aptos/src/governance/mod.rs:169]: group yeah only lets you have one or the other

**Review by movekevin — APPROVED** [review]

File: crates/aptos/src/governance/mod.rs [hunk 2]
```diff
@@ -83,53 +87,31 @@ impl CliCommand<Transaction> for SubmitProposal {
...
-        })?;
-        Url::parse(&metadata.discussion_url).map_err(|err| {
-            CliError::CommandArgumentError(format!(
-                "Discussion URL {} is invalid {}",
-                metadata.discussion_url, err
-            ))
-        })?;
+        // Check script file
+        let script_path = self.script_path.as_path();
+        if self.script_path.exists() {
```
Comment (movekevin) [inline crates/aptos/src/governance/mod.rs:92]: I feel like we do these 2 checks enough that they should be refactored into a util function or something

File: crates/aptos/src/governance/mod.rs [hunk 3]
```diff
@@ -220,79 +255,106 @@ impl std::fmt::Display for ProposalMetadata {
...
+        prompt_options,
+    )?;
+
+    // Insert the new script
+    let sources_dir = package_dir.join("sources");
+    let new_script_path = if let Some(file_name) = script_path.file_name() {
+        sources_dir.join(file_name)
+    } else {
+        // If for some reason we can't get the move file
+        sources_dir.join("script.mv")
```
Comment (movekevin) [inline crates/aptos/src/governance/mod.rs:284]: script.move?

File: crates/aptos/src/governance/mod.rs [hunk 4]
```diff
@@ -352,17 +417,28 @@ impl CliCommand<TransactionSummary> for ExecuteProposal {
...
 
     async fn execute(mut self) -> CliTypedResult<TransactionSummary> {
-        if !self.path.exists() {
-            return Err(CliError::UnableToReadFile(
-                self.path.display().to_string(),
-                "Path doesn't exist".to_string(),
-            ));
+        // TODO: Make one object so this code isn't copied
+        // Check script file
+        let script_path = self.script_path.as_path();
```
Comment (movekevin) [inline crates/aptos/src/governance/mod.rs:422]: All logic here and below seems redundant with same logic in create proposal. Can we refactor them into a separate func?

**Review by movekevin — APPROVED** [review]
Flow looks good to me

File: crates/aptos/src/governance/mod.rs (hunk 2 above)
Comment (gregnazario) [reply crates/aptos/src/governance/mod.rs:92]: Yeah, I'll just package them up now

**Pull request merged**